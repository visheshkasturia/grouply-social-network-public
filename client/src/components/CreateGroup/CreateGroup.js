/*eslint-disable */
import { useState } from "react";
import PostGroups from "../../apiCalls/PostGroups";

export default function CreateGroup( {user} ) {
  const [inputs, setInputs] = useState({});

  const handleChange = (event) => {
    const name = event.target.name;
    const value = event.target.value;
    setInputs(values => ({...values, [name]: value}))
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    inputs.createdBy = user._id;
    PostGroups(inputs);
    setInputs({});
  }

  return (
    <form onSubmit={handleSubmit}>
      <label>Enter Group Name:
      <input 
        type="text" 
        name="groupName" 
        value={inputs.groupName || ""} 
        onChange={handleChange}
      />
      <br />
      </label>
      <label>Enter Group description:
      <input 
        type="text" 
        name="description" 
        value={inputs.description || ""} 
        onChange={handleChange}
      />
      </label>
      <br />
      <label>Enter Group Privacy:
        <input 
          type="text" 
          name="privacy" 
          value={inputs.privacy|| ""}
          placeholder="0 for public, 1 for private" 
          onChange={handleChange}
        />
        </label>
        <br />
        <label>Enter Group Tags:
        <input 
          type="text" 
          name="tags" 
          value={inputs.tags|| ""} 
          onChange={handleChange}
        />
        </label>
        <br />
        <button type="submit"> Submit </button> 
    </form>
  )
}

// ReactDOM.render(<MyForm />, document.getElementById('root'));