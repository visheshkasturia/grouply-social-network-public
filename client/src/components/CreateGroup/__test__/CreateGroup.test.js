import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CreateGroup from '../CreateGroup'

it('CreateGroup renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CreateGroup />, div);
});

it('CreateGroup matches snapshot', () => {
  const tree = renderer.create(<CreateGroup />).toJSON();
  expect(tree).toMatchSnapshot();
});