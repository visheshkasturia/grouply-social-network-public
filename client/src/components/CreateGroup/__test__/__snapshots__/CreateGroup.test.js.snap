// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGroup matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Enter Group Name:
    <input
      name="groupName"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
  </label>
  <label>
    Enter Group description:
    <input
      name="description"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    Enter Group Privacy:
    <input
      name="privacy"
      onChange={[Function]}
      placeholder="0 for public, 1 for private"
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    Enter Group Tags:
    <input
      name="tags"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <button
    type="submit"
  >
     Submit 
  </button>
</form>
`;
